app/jobs/elasticsearch_delete_by_query_job.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+# Deletes documents from the Elasticsearch index where "updated_at" field is older than the configured retention days
+# - Uses async delete_by_query and polls task progress
+# - Configurable throttling and defaults
+
+class ElasticsearchDeleteByQueryJob < ApplicationJob
+  queue_as :searchgov
+
+  DEFAULT_REQUESTS_PER_SECOND = 500
+  DEFAULT_SCROLL_SIZE = 5000
+  DEFAULT_MAX_TASK_WAIT_SECONDS = 23 * 60 * 60 # 23h
+  START_RETRY_ATTEMPTS = 3
+
+  def perform
+    index = fetch_env!('SEARCHELASTIC_INDEX')
+    retention_days = fetch_retention_days!
+
+    Rails.logger.info { "ElasticsearchDeleteByQueryJob starting for index=#{index}, retention_days=#{retention_days}" }
+
+    body = {
+      query: {
+        range: {
+          updated_at: { lt: "now-#{retention_days}d/d" }
+        }
+      }
+    }
+
+    requests_per_second = DEFAULT_REQUESTS_PER_SECOND
+    scroll_size = DEFAULT_SCROLL_SIZE
+    max_wait_seconds = DEFAULT_MAX_TASK_WAIT_SECONDS
+
+    slices = 'auto'
+
+    response = with_start_retries(index, body, slices, requests_per_second, scroll_size)
+
+    task_id = (response.is_a?(Hash) && response['task']) ? response['task'] : nil
+
+    if task_id.nil?
+      deleted = (response && response['deleted']) || 0
+      Rails.logger.info { "delete_by_query completed synchronously; deleted=#{deleted} index=#{index}" }
+      return
+    end
+
+    Rails.logger.info { "delete_by_query started async task_id=#{task_id} index=#{index} slices=#{slices} rps=#{requests_per_second}" }
+
+    begin
+      result = wait_for_task_completion(task_id, max_wait_seconds)
```

Comment:
Why do we need to wait for this to complete like this? What's the issue with just let `ES.client.delete_by_query` do it's thing. This might be the right thing to do but it just out to me because it's not a common/advisable ruby pattern to wrap things in a loop like this.

Reply:
I see. Apparently there is a `wait_for_completion` flag you can pass to `delete_by_query` and just let the job wait until it finish instead of doing this async. It's Ok to wait for it since it's happening in a background job. This way we'll see the job is still "working" and we could tell deletion is still happening in the resque ui. Can you pass this flag and just let the library to handle the waiting?

https://www.elastic.co/guide/en/elasticsearch/reference/7.17/docs-delete-by-query.html#docs-delete-by-query-task-api